ingestion/src/main/java/feast/ingestion/transform/metrics/WriteMetricsTransform.java

```diff
@@ -49,13 +59,18 @@ public static Builder newBuilder() {
 
     public abstract Builder setFailureTag(TupleTag<FailedElement> failureTag);
 
+    public abstract Builder setFeatureSetByRef(
+        Map<String, FeatureSetProto.FeatureSet> featureSetByRef);
+
     public abstract WriteMetricsTransform build();
   }
 
   @Override
   public PDone expand(PCollectionTuple input) {
     ImportOptions options = input.getPipeline().getOptions().as(ImportOptions.class);
-    switch (options.getMetricsExporterType()) {
+    assert options.getMetricsExporterType() != null;
```

Comment:
What happens when this assertion fails? How can the user react?

Reply:
This assertion I put here is meant to catch our programming error not for the user to handle.
The default currently is set to "none" string and it should never be `null` unless there is some error in the codes.
https://github.com/davidheryanto/feast/blob/update-ingestion-metrics-for-validation/ingestion/src/main/java/feast/ingestion/options/ImportOptions.java#L72